docs/organizations/invitations.md

```diff
@@ -14,10 +14,24 @@ Invitees may or may not be existing users. If you choose to allow invitations to
 
 It is possible to pre-select any roles for a user by passing the `roles` array in the API (pre-selecting in the Admin UI is currently not implemented). This will automatically add the user to the given roles upon user creation and acceptance of the invitation.
 
-Invitees will receive an email indicating the realm, organization and inviter. It is possible to add a specific application redirect URI so that they will be redirected to a specific location in the application.
+Make to adapt the [Authentication](https://github.com/p2-inc/keycloak-orgs?tab=readme-ov-file#authentication) according to your needs. Note email verification must be enabled.
 
 Following user creation and authentication, invitees will be prompted to accept or decline any outstanding invitations. In addition to any roles that were selected in the invitation, the user will automatically be added as a member to the inviting organization.
 
+### Options for user registration
+
+#### Self-registration
+
+This option requires either creating users on behalf of them before sending the invitation or enabling user registration on the realm.
+
+Invitees will receive an email indicating the realm, organization and inviter. It is possible to add a specific application redirect URI so that they will be redirected to a specific location in the application.
+
+#### Magic-Link
+
+This approach is prefered when user registration is disabled. In this flow the invitation would have `send` set to false to prevent sending an email immediately. The invitee would then be send an email outside of Keycloak/phaseTwo with a [magic link](/docs/authentication/magic-links). The magic link redirect URI takes precedence over any redirect URI set on the invitation.
```

Comment:
I think this is somewhat ambiguous. How about:

This approach uses another extension in order to provide a [magic link](https://github.com/p2-inc/keycloak-magic-link) that will first register the user, authenticate them, and then present them with the required action to accept the invitation. This approach is preferred when user registration is disabled. It requires the following steps:
1. [Create a magic link](https://phasetwo.io/api/create-magic-link) using `force_create=true` (to create a user if one does not exist) and `send_email=false` to skip sending the magic link to the user.
2. Use the returned magic link when [creating an invitation](https://phasetwo.io/api/add-organization-invitation). Set the `redirectUri={magic-link}`.

Reply:
updated as per your suggestions